go/api/base_client.go

```diff
@@ -4926,7 +4930,11 @@ func (client *baseClient) RestoreWithOptions(key string, ttl int64,
 	if err != nil {
 		return CreateNilStringResult(), err
 	}
-	return handleStringOrNilResponse(result)
+	response, err := handleOkResponse(result)
+	if err != nil {
+		return CreateNilStringResult(), err
+	}
+	return CreateStringResult(response), nil
```

Comment:
it wasn't what I meant - I meant that the response has to be Ok or Error, it shouldn't be nil in any case. Does `return handleOkResponse(result)` break some tests?